packages/plugin-commands-env/src/env.ts

```diff
@@ -173,6 +191,45 @@ export async function handler (opts: NvmNodeCommandOptions, params: string[]) {
     return `Node.js ${nodeVersion} is removed
   ${versionDir}`
   }
+  case 'list':
+  case 'ls': {
+    if (!opts.global && !opts.cliOptions?.remote) {
+      throw new PnpmError('NOT_IMPLEMENTED_YET', '"pnpm env list <option>" can only be used with the "--global" option or "--remote" option currently')
+    }
+    if (opts.cliOptions?.remote) {
+      const fetch = createFetchFromRegistry(opts)
+      const { releaseChannel, versionSpecifier } = parseNodeEditionSpecifier(params[1] ?? '')
+      const nodeMirrorBaseUrl = getNodeMirror(opts.rawConfig, releaseChannel)
+      const nodeVersionList = await resolveNodeVersionList(fetch, versionSpecifier, nodeMirrorBaseUrl)
+      // make the newest version located in the end of output
+      return nodeVersionList.reverse().join('\n')
+    }
+    const nodeDir = getNodeVersionsBaseDir(opts.pnpmHomeDir)
+    const nodePath = path.resolve(opts.pnpmHomeDir, process.platform === 'win32' ? 'node.exe' : 'node')
+    let currentNodeVersion: string | undefined
+    let nodeLink: string | undefined
+
+    if (!existsSync(nodeDir)) {
+      throw new PnpmError('ENV_NO_NODE_DIRECTORY', `Couldn't find Node.js directory in ${nodeDir}`)
+    }
+
+    try {
+      nodeLink = await fs.readlink(nodePath)
+    } catch (err) {
+      nodeLink = undefined
+    }
+
+    const nodeVersionDirs = await fs.readdir(nodeDir)
+    const nodeVersions = nodeVersionDirs.filter(nodeVersion => {
+      const nodeSrc = path.join(nodeDir, nodeVersion, process.platform === 'win32' ? 'node.exe' : 'bin/node')
+      const nodeVersionDir = path.join(nodeDir, nodeVersion)
+      if (nodeLink?.includes(nodeVersionDir)) {
+        currentNodeVersion = nodeVersion
+      }
+      return semver.valid(nodeVersion) && existsSync(nodeSrc)
+    })
+    return nodeVersions.map(nodeVersion => `${nodeVersion === currentNodeVersion ? '*' : ' '} ${nodeVersion}`).join('\n')
```

Comment:
Move this logic to a new file called `envList.ts`

Reply:
Devide it to two functions. listRemoveVersions and listLocalVersions

Reply:
Ok